src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -222,7 +222,7 @@ private static object ValueFromScalar(ScalarGraphType scalar, object input)
 
             var values = new Dictionary<string, object>(definitionArguments.Count);
 
-            foreach (var arg in definitionArguments.ArgumentsList)
+            foreach (var arg in definitionArguments)
```

Comment:
`ArgumentsList` was used to not allocate iterator object in the heap. `List<T>.Iterator` is struct.

Reply:
It's only used when there are arguments, but still, I add an optimization

Reply:
It's not worth making an optimization here (or retaining the existing one).  I've ran benchmarks over and over with different techniques.  I can't do better than 0.1% less memory allocated (0.5 KB) with no measurable speed improvement or GC effect on the introspection test.  For the hero test, it shows about 1.5% speed improvement and no GC or memory difference at all.

The code just isn't on the hot path, as it only executes specifically when a field resolver calls `GetArgument<>`, and even then, only once per resolver.  Sometimes the simplest answer is the best.

See the `ReadonlyResolveFieldContext.Arguments` property, which lazy loads the argument values

If you want me to optimize it anyway, I can unroll the foreach loop into two separate copies, one when `definitionArguments is QueryArguments` and one otherwise, and after that, I can optionally consolidate the inner code with a local function or private class function.  The compiler can prevent heap allocations when using local functions rather than lambda functions (which always require a heap allocation).

See: https://docs.microsoft.com/en-us/dotnet/csharp/local-functions-vs-lambdas

Reply:
can be left unchanged